storybookjs/storybook
# [SIGNIFICANT] Remove redundant render in react

Issue:

In the React framework code, `ReactDOM.render` would get unnecessarily called twice.

## What I did

Exit early from the render function if nothing is changing.

## How to test

I added a story "Lifecycle:logging" to the cra-kitchen-sink app that demostrates the problem:

To reproduce: browse to the story, hit reload in the browser (as it happens whenever the preview redux store changes, there are a probably other ways to reproduce).

**Previously**: the component would re-render, which would log extra lines to the console.

**Now**: it simply renders once.

We should also satisfy ourselves that no other feature was relying on this bug (cf https://github.com/storybooks/storybook/pull/2463)

### Is this testable with jest or storyshots?

Not right now; it would be good to maybe use JSDOM to test this kind of thing.

### Does this need a new example in the kitchen sink apps?

Yes, done

### Does this need an update to the documentation?

No

If your answer is yes to any of these, please make sure to include it in your PR.
